google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -103,15 +106,33 @@ public static EnhancedBigtableStub create(EnhancedBigtableStubSettings settings)
   }
 
   // <editor-fold desc="Callable creators">
+
+  /**
+   * Creates a callable chain to handle ReadRows RPCs. The chain will:
+   *
+   * <ul>
+   *   <li>Convert a {@link Query} into a {@link com.google.bigtable.v2.ReadRowsRequest} and
+   *       dispatch the RPC.
+   *   <li>Upon receiving the response stream, it will merge the {@link
+   *       com.google.bigtable.v2.ReadRowsResponse.CellChunk}s in logical rows. The actual row
+   *       implementation can be configured in {@link
+   *       com.google.cloud.bigtable.data.v2.BigtableDataSettings}.
+   *   <li>Retry/resume on failure.
+   *   <li>Filter out marker rows.
+   * </ul>
+   */
   public <RowT> ServerStreamingCallable<Query, RowT> createReadRowsCallable(
       RowAdapter<RowT> rowAdapter) {
-    return new ServerStreamingCallable<Query, RowT>() {
-      @Override
-      public void call(
-          Query query, ResponseObserver<RowT> responseObserver, ApiCallContext context) {
-        throw new UnsupportedOperationException("todo");
-      }
-    };
+
+    ServerStreamingCallable<ReadRowsRequest, RowT> merging =
+        new RowMergingCallable<>(stub.readRowsCallable(), rowAdapter);
+
+    FilterMarkerRowsCallable<RowT> filtering = new FilterMarkerRowsCallable<>(rowAdapter, merging);
+
+    ServerStreamingCallable<ReadRowsRequest, RowT> withContext =
+        filtering.withDefaultCallContext(clientContext.getDefaultCallContext());
+
+    return new ReadRowsUserCallable<>(withContext, requestContext);
```

Comment:
Actually I have one more comment - the default call context should be added to the outer-most callable.

Reply:
Unfortunately I can't do that because of the special handling for `first()`.

AnonymousWithDefaultCallContextCallable#first() will return a FirstElementCallable,
whose call() will invoke AnonymousWithDefaultCallContextCallable#futureCall() which will bypass any custom FirstCallable I might set

Reply:
Hmm that kind of bothers me - should we have a more first-class way of overriding first() that works in this case too? Otherwise, other implementers could get tricked because they didn't realize this fact.

Reply:
I think this would need to be addressed in gax. Would you be ok with continuing this discussion in an issue on gax-java but merging this in the mean time?

Reply:
I can also remove this optimization until we figure out a better solution. Let me know which works for you

Reply:
No I think retaining the optimization is good, otherwise we might forget to resolve the design issue and might never have the optimization. Could you put a note here about why the default call context can't be added to the outermost callable, and that resolving it requires design changes in gax?